scrapers/in/apiclient.py

```diff
@@ -89,11 +89,16 @@ def get_session_no(self, session):
         headers["Accept"] = "application/json"
         headers["User-Agent"] = self.user_agent
         url = urljoin(self.root, f"/{session}")
-        resp = requests.get(url, headers=headers).json()
-        session_no_regex = re.search(r"Session\s+(\d+).+", resp["name"])
 
-        if session_no_regex:
-            session_no = session_no_regex.group(1)
+        while not session_no:
```

Comment:
I'm a little worried that we need some kind of backoff logic in here, so that we don't risk having a scraper that indefinitely repeats a request to the source website without any delay between requests. I think there's an example in `ga/util.py`.

We need to avoid creating a scenario where a scraper can hammer a source site with a (possibly) infinite loop of requests.

Reply:
@jessemortenson Thanks for your review. I added a backoff function as you mentioned. Please review it.